### PR TITLE
Azure Environment Provider to retrieve Failure Domain Info from Insta…

### PR DESCRIPTION
**Pull request by desaijay230592**
…nce Metadata Service and configure custom Pinot Configuration for updating server instance config in zookeeper node

## Description
Creating generic interface PinotEnvironment which can be implemented by various cloud providers to customize the PinotConfiguration by setting different environment variables and other instance specific configuration. 

Primary aim is to update the server instance config with failure/fault domain information which will be useful when designing segment assignment strategy to achieve high availability. 

This change introduces a PinotEnvironment interface under pinot-spi(a.ka pinot-service-provider-interface). We introduce a pinot-enviroment/pinot-azure  plugin containing AzureEnvironmentProvider class to provide azure specific pinot configuration customization which contains failure domain information. This failure domain information is retrieved from Azure IMDS (Instance Metadata Service) wherein call has to be made from within the VM. 

Unit Tests are written for the same. End-End integration tests are in-progress. Will take LinkedIN Pinot-SRE help to conduct those.

Design Document: https://docs.google.com/document/d/1echXSSZfgKWLtkmA1KKxmdCRUkTqffqoJhNA-ovTfzc/ 

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
   public static final String PREFIX_OF_SSL_SUBSET = "ssl";
   public static final String HTTP_PROTOCOL = "http";
   public static final String HTTPS_PROTOCOL = "https";
+  public static final String INSTANCE_FAILURE_DOMAIN = "pinot.environment.instance.failureDomain";
```
Comment (rkanumul) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java:29]: IMO, Better place for this is in the PinotEnvironment interface. to have properties located with the interface itself.

File: pinot-plugins/pinot-environment/pom.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  <parent>
+    <artifactId>pinot-plugins</artifactId>
+    <groupId>org.apache.pinot</groupId>
+    <version>0.7.0-SNAPSHOT</version>
+    <relativePath>..</relativePath>
+  </parent>
+
+  <artifactId>pinot-environment</artifactId>
+  <packaging>pom</packaging>
+  <name>Pluggable Pinot file system </name>
```
Comment (rkanumul) [inline pinot-plugins/pinot-environment/pom.xml:35]: Please update description

File: pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java (hunk 1 above)
Comment (desaijay230592) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java]: Updated.

File: pinot-plugins/pinot-environment/pom.xml (hunk 2 above)
Comment (desaijay230592) [reply pinot-plugins/pinot-environment/pom.xml]: Updated.

**Pull request closed without merging**